src/CLIEngine.js

```diff
@@ -197,12 +226,13 @@ const getFileList = function(patterns, options) {
 
   const files = [];
   const addedFiles = new Set();
+  const ignorer = getIgnorer(cwd, options);
 
   globPatterns.forEach((pattern) => {
     const file = path.resolve(cwd, pattern);
 
     if (fs.existsSync(file) && fs.statSync(file).isFile()) {
-      if (addedFiles.has(file)) {
+      if (addedFiles.has(file) || ignorer.ignores(pattern)) {
```

Comment:
It looks like `ignorer.ignores()` takes a path name. I wonder if it is having trouble accepting a glob pattern. What do you think about passing in `file` instead. `file` is getting resolved based on the `cwd` and the `pattern` on line 232.